### ISSUE TITLE
crash in JacobianMorphism_divisor_class_field when variable is not called x

### ISSUE THREAD

**Issue by yyyyx4** [sagemath/sage#30087]
<div id="comment:0"></div>

The following code:

```python
R.<v> = QQ[]
C = HyperellipticCurve(v**3-1)
J = C.jacobian()
D = J(v-1,0)
```

crashes on sage version 9.1 and the current `develop` branch with the exception:

```
Traceback (most recent call last):
  File "./mfe.sage.py", line 10, in <module>
    D = J(v-_sage_const_1 ,_sage_const_0 )
  File "sage/schemes/generic/scheme.py", line 265, in __call__
    return self.point(args)
  File "sage/schemes/hyperelliptic_curves/jacobian_generic.py", line 147, in point
    return self(self.base_ring())(mumford)
  File "schemes/hyperelliptic_curves/jacobian_homset.py", line 149, in __call__
    return JacobianMorphism_divisor_class_field(self, (P1, P2))
  File "sage/schemes/hyperelliptic_curves/jacobian_morphism.py", line 386, in __init__
    if not (b**2 + h*b - f)%a == 0:
  File "sage/structure/element.pyx", line 1517, in sage.structure.element.Element.__mul__ (build/cythonized/sage/structure/element.c:12231)
  File "sage/structure/coerce.pyx", line 1255, in sage.structure.coerce.CoercionModel.bin_op (build/cythonized/sage/structure/coerce.c:11275)
TypeError: unsupported operand parent(s) for *: 'Univariate Polynomial Ring in v over Rational Field' and 'Univariate Polynomial Ring in x over Rational Field'
```

The reason is that the code uses `x` internally and doesn't expect the input to come from a "different" univariate polynomial ring, hence the exception. The error does not occur when `v` is replaced by `x` in the example above.

The proposed patch explicitly casts the given polynomials to the parent ring of the curve's defining polynomials.

CC:  @alexjbest @kedlaya @slel

Component: **algebraic geometry**

Keywords: **hyperelliptic jacobian**

Author: **Lorenz Panny**

Branch/Commit: **[public/fix_hyperelliptic_jacobian_crash_variable_name](https://github.com/sagemath/sagetrac-mirror/tree/public/fix_hyperelliptic_jacobian_crash_variable_name) @ [`627adb0`](https://github.com/sagemath/sagetrac-mirror/commit/627adb0ae77595b534d2ffa14ebfbedac4883bca)**

_Issue created by migration from https://trac.sagemath.org/ticket/30087_

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`a9f3042`](https://github.com/sagemath/sagetrac-mirror/commit/a9f30428b6c99db1b258d171b8e4e45965013a95)** to **[`627adb0`](https://github.com/sagemath/sagetrac-mirror/commit/627adb0ae77595b534d2ffa14ebfbedac4883bca)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:23"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/627adb0ae77595b534d2ffa14ebfbedac4883bca"><code>627adb0</code></a></td><td><code>Merge tag '9.8.beta2' into public/fix_hyperelliptic_jacobian_crash_variable_name</code></td></tr></table>

**Comment by vincentmacri** [comment]
Fixed in 10.5 beta4, I believe by #38459.
